compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEInsertLoopsForVectorization.cpp

```diff
@@ -43,11 +44,31 @@ class AMDAIEInsertLoopsForVectorizationPass
     return nDims - firstDim;
   };
 
+  /// Tile the generic op using `tileSizes` and coalesce the generated tiling
+  /// loops in order to minimize the overhead of loop control/branch statements.
+  static void performTiling(IRRewriter &rewriter,
+                                     linalg::GenericOp genericOp,
+                                     SmallVector<int64_t> &tileSizes,
+                                     bool isTensorType) {
+    auto opts = linalg::LinalgTilingOptions().setTileSizes(tileSizes);
+    auto tiled = linalg::tileLinalgOp(rewriter, genericOp, opts);
+    const auto &tileLoops = tiled.value().loops;
+    SmallVector<scf::ForOp> loops = llvm::map_to_vector(
+        tileLoops, [](Operation *loop) { return cast<scf::ForOp>(loop); });
+    (void)mlir::coalesceLoops(rewriter, loops);
+    if (isTensorType) {
+      rewriter.replaceOp(genericOp, loops[0]->getResult(0));
+    } else {
+      rewriter.eraseOp(genericOp);
```

Comment:
Could you document that this function can work with both tensors and memrefs above?